client/clientservice/src/client_service.cpp

```diff
@@ -13,13 +13,17 @@
 
 #include <chrono>
 #include <grpcpp/grpcpp.h>
+#include <grpcpp/health_check_service_interface.h>
+#include <grpcpp/ext/proto_server_reflection_plugin.h>
+#include <grpcpp/ext/health_check_service_server_builder_option.h>
 
 using namespace std::literals::chrono_literals;
 
 namespace concord::client::clientservice {
 
 void ClientService::start(const std::string& addr, unsigned num_async_threads, uint64_t max_receive_msg_size) {
-  grpc::EnableDefaultHealthCheckService(true);
```

Comment:
did we review if there is a way to extend or register callbacks into the default health check service implementation? Seems their code is well structured but not sure the classes are accessible for external modules.

Reply:
@bsenthilr we are deriving from the default health check service, pls refer to https://github.com/vmware/concord-bft/pull/2819/files#diff-7ede078b0f0a07554b6fd9358bebe4c37b3dec7d492cd0a899e62e0d0acf3d5f -

```
// A sync implementation of the health checking service.
class HealthCheckServiceImpl : public grpc::health::v1::Health::Service {
 public:
...
}
```

We needed to be able to query client-pool -> replica connections everytime `Check`/`Watch` are called, hence these methods had to be overridden and custom implementations added.